ochre/utils/hpxml.py

```diff
@@ -475,36 +495,37 @@ def parse_hpxml_zones(hpxml, boundaries, construction):
-        if not has_garage and len(attic_wall_areas) == 2:
+        if len(attic_wall_areas) == 2:
+            # standard gable roof with attic
             assert abs(attic_wall_areas[1] - attic_wall_areas[0]) < 0.2  # computational errors possible
             attic_gable_area = attic_wall_areas[0]
-            garage_gable_area = 0
+            third_gable_area = 0
         elif has_garage and len(attic_wall_areas) == 3:
-            # 2 attic gables, 1 garage gable, garage gable has area that is 'more different'
+            # 2 attic gables plus 1 garage gable, garage gable has area that is 'more different'
             attic_gable_area = attic_wall_areas[1]
             low, med, high = tuple(sorted(attic_wall_areas))
-            garage_gable_area = low if med - low > high - med else high
+            third_gable_area = low if med - low > high - med else high
         else:
-            raise Exception('Unable to parse gable wall areas.')
+            raise Exception('Unable to calculate attic area, likely an issue with gable walls.')
 
         # Get attic properties
-        roof_tilt = convert(boundaries['Attic Roof']['Tilt (deg)'], 'deg',
-                            'rad')  # tan(roof_tilt) = height / (width / 2)
         attic_height = (attic_gable_area * math.tan(roof_tilt)) ** 0.5
-        attic_volume = 1 / 2 * second_floor_area * attic_height  # volume = 1/2 l*w*h
-        if garage_gable_area > 0:
+        if third_gable_area > 0:
             # assumes a combined attic, add volume over garage and in between
-            garage_tilt = convert(boundaries['Garage Roof']['Tilt (deg)'], 'deg', 'rad') \
-                if 'Garage Roof' in boundaries else roof_tilt
-            garage_height = (garage_gable_area * math.tan(garage_tilt)) ** 0.5
-            garage_width = 2 * garage_gable_area / garage_height
+            square_area = attic_floor_area - garage_protruded_area
+            garage_height = (third_gable_area * math.tan(garage_tilt)) ** 0.5
+            garage_width = 2 * third_gable_area / garage_height
             # garage_length = garage_floor_area / garage_width
-            garage_depth_in_house = garage_height * math.tan(roof_tilt)  # length from ext wall to roof connection point
-            attic_volume += (1 / 2 * garage_floor_area * garage_height +  # area over garage
-                             1 / 6 * garage_width * garage_depth_in_house * garage_height)  # pyramid between prisms
+            garage_depth_in_house = garage_height * math.tan(roof_tilt)  # length from attached wall to roof connection point
+            attic_volume = (1 / 2 * square_area * attic_height +                           # prism over square
+                            1 / 2 * garage_protruded_area * garage_height +                # prism over garage
+                            1 / 6 * garage_width * garage_depth_in_house * garage_height)  # pyramid between prisms
```

Comment:
Should this be there if it's a 2 story home with attic over garage and then the attic is the 3rd story? I think no?

Reply:
This only happens if the attic isn't square, i.e. there's 3 gable walls. That can happen in a 2 story home if the 2nd floor is fully over the garage.